homeassistant/components/google_generative_ai_conversation/conversation.py

```diff
@@ -0,0 +1,162 @@
+"""Conversation support for the Google Generative AI Conversation integration."""
+
+from __future__ import annotations
+
+import logging
+from typing import Literal
+
+from google.api_core.exceptions import ClientError
+import google.generativeai as genai
+import google.generativeai.types as genai_types
+
+from homeassistant.components import assist_pipeline, conversation
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import MATCH_ALL
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import TemplateError
+from homeassistant.helpers import intent, template
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.util import ulid
+
+from .const import (
+    CONF_CHAT_MODEL,
+    CONF_MAX_TOKENS,
+    CONF_PROMPT,
+    CONF_TEMPERATURE,
+    CONF_TOP_K,
+    CONF_TOP_P,
+    DEFAULT_CHAT_MODEL,
+    DEFAULT_MAX_TOKENS,
+    DEFAULT_PROMPT,
+    DEFAULT_TEMPERATURE,
+    DEFAULT_TOP_K,
+    DEFAULT_TOP_P,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up conversation entities."""
+    agent = GoogleGenerativeAIConversationEntity(hass, config_entry)
+    async_add_entities([agent])
+
+
+class GoogleGenerativeAIConversationEntity(
+    conversation.ConversationEntity, conversation.AbstractConversationAgent
+):
+    """Google Generative AI conversation agent."""
+
+    _attr_has_entity_name = True
```

Comment:
OpenAI doesn't have this.

Reply:
My impression is that this is an additional omission from the Open AI PR and that this is mandatory: https://developers.home-assistant.io/docs/core/entity#has_entity_name-true-mandatory-for-new-integrations

Open AI doesn't set a device so its a no-op, but I think it should.